### ISSUE TITLE
Invalid selector errors are not catched

### ISSUE THREAD

**Issue by bcalou** [gilbarbara/react-joyride#1035]
Hi and thanks for your work!

## 🐛 Bug Report

When using a "modern" CSS selector such as :has, older browser will fail to use `document.querySelector`, which is expected :

```
Uncaught DOMException: Document.querySelector: '.my-form__field:has(input#name)' is not a valid selector
```

However, the error is not catched and the whole app crashes (white page).

## To Reproduce

Use the reproduction link below or : 
Try to select an element with a selector that your browser does not support.
You can make up a selector that doesn't exist so it will fail regardless of the browser.

## Expected behavior

I believe the error should be catched and the step (or tour) skipped.
I think it may be a common mistake to think that the selector will fail silently (as they do when using CSS, but not when using querySelector).

## Link to repl or repo (highly encouraged)

Try this tour : the second step should crash your page.

[codesandbox](https://codesandbox.io/p/devbox/pedantic-surf-yv9nh6?workspaceId=3a9cd981-5235-42d1-9693-a8c216b7bd18&layout=%257B%2522sidebarPanel%2522%253A%2522SEARCH%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clwrnvri900072869m3b3gg1e%2522%252C%2522sizes%2522%253A%255B59.7607122982749%252C40.2392877017251%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clwrnvri900022869frtgvckn%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clwrnvri9000428698995548r%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clwrnvri900062869be8hhr64%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clwrnvri900022869frtgvckn%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clwrnvri900012869s6opy4oh%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clwrnwa3b00542869o73532go%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clwrnvri900022869frtgvckn%2522%252C%2522activeTabId%2522%253A%2522clwrnwa3b00542869o73532go%2522%257D%252C%2522clwrnvri900062869be8hhr64%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clwrnvri9000528691ryk6td7%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522start%2522%252C%2522port%2522%253A3000%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clwrnvri900062869be8hhr64%2522%252C%2522activeTabId%2522%253A%2522clwrnvri9000528691ryk6td7%2522%257D%252C%2522clwrnvri9000428698995548r%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clwrnvri900032869ew2r4lj2%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522start%2522%257D%255D%252C%2522id%2522%253A%2522clwrnvri9000428698995548r%2522%252C%2522activeTabId%2522%253A%2522clwrnvri900032869ew2r4lj2%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

Thanks :)

**Comment by gilbarbara** [comment]
Hey @bcalou 
Thanks for your report. I've already fixed it in the #1036 PR. I'm waiting for a response to another issue, and I'll release it next week.

**Comment by gilbarbara** [comment]
Fixed in 2.9.0

Thanks
